starlite/handlers/http_handlers.py

```diff
@@ -497,36 +504,53 @@ def resolve_response_class(self) -> Type["Response"]:
                 return layer.response_class
         return Response
 
-    def resolve_response_headers(self) -> "ResponseHeadersMap":
+    def resolve_response_headers(self) -> FrozenSet["ResponseHeader"]:
         """Return all header parameters in the scope of the handler function.
 
         Returns:
             A dictionary mapping keys to :class:`ResponseHeader <starlite.datastructures.ResponseHeader>` instances.
         """
         resolved_response_headers: dict[str, "ResponseHeader"] = {}
+
         for layer in self.ownership_layers:
-            resolved_response_headers.update(layer.response_headers or {})
+            if layer_response_headers := layer.response_headers:
+                if isinstance(layer_response_headers, Mapping):
```

Comment:
I'm unsure if this is the right approach. This can't happen unless you modify a handler instance, which isn't a case we necessarily need to cover IMO, but from a type checking perspective, this is correct, since `Controller` doesn't differentiate between the type given in `__init__`  and the one set on the instance.  Maybe we should just raise an exception in that case, or `# type: ignore` it?